developers/weaviate/client-libraries/python/async.md

```diff
@@ -0,0 +1,196 @@
+
+## Instantiation
+
+An async client `WeaviateAsyncClient` object can be instantiated [using a helper function](#instantiation-helper-functions), or by [explicitly creating an instance of the class](#direct-instantiation).
+
+### Instantiation helper functions
+
+These instantiation helper functions mirror the [synchronous client helper functions](./index.md#connection-helper-functions), and return an equivalent async client object.
+
+- `use_async_with_local`
+- `use_async_with_weaviate_cloud`
+- `use_async_with_custom`
+
+However, the async helper functions do not connect to the server as their synchronous counterparts do.
+
+When using the async helper functions, you must call the async `.connect()` method to connect to the server, and call `.close()` before exiting to clean up. (Except when using a [context manager](#context-manager).)
+
+The async helper functions take the same parameters for [external API keys](./index.md#external-api-keys), [connection timeout values](./index.md#timeout-values) and [authentication details](./index.md#authentication).
+
+<Tabs groupId="languages">
+<TabItem value="wcd" label="WCD">
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# AsyncWCDInstantiation"
+  endMarker="# END AsyncWCDInstantiation"
+  language="py"
+/>
+
+</TabItem>
+<TabItem value="local" label="Local">
+
+  <FilteredTextBlock
+    text={PythonCode}
+    startMarker="# AsyncLocalInstantiationBasic"
+    endMarker="# END AsyncLocalInstantiationBasic"
+    language="py"
+  />
+
+</TabItem>
+
+<!-- TODO - add embedded equivalent when available in client -->
+
+<TabItem value="custom" label="Custom">
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# AsyncCustomInstantiationBasic"
+  endMarker="# END AsyncCustomInstantiationBasic"
+  language="py"
+/>
+
+</TabItem>
+</Tabs>
+
+### Explicit instantiation
+
+If you need to pass custom parameters, use the `weaviate.WeaviateAsyncClient` class to instantiate a client. This is the most flexible way to instantiate the client object.
+
+When you instantiate a connection directly, you have to call the (now async) `.connect()` method to connect to the server.
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# AsyncDirectInstantiationFull"
+  endMarker="# END AsyncDirectInstantiationFull"
+  language="py"
+/>
+
+## Sync and async methods
+
+The async client object is designed to be used in an `async` function or in an [`asyncio` event loop](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio-event-loop).
+
+Accordingly, a majority of the client methods are `async` functions that return [`Coroutines` objects](https://docs.python.org/3/library/asyncio-task.html#coroutine). However, some methods are synchronous and can be used in a synchronous context.
+
+As a rule of thumb, a method that involves a request to Weaviate is likely to be an async function, while a method that executes in a local context is likely to be synchronous.
```

Comment:
I think you can be stronger here in the language and say `will` instead of `likely` as all I/O requests must be `async` functions by nature. Likewise, involving the event loop for in-memory CPU work is highly inefficient (I saw this born out during development) so in-mem compute will always be `sync`